libs/sr/formula/src/data/char/RuanMei.ts

```diff
@@ -16,54 +25,141 @@ import { dmg, entriesForChar, getBaseTag, scalingParams } from './util'
+    delayFromBreak: ult[0][u++],
+    breakDmg: ult[0][u++],
   },
   talent: {
-    dmg: talent[0][ta++],
+    spd_: talent[0][ta++],
+    breakDmg: talent[0][ta++],
   },
   technique: {
-    dmg: (technique[0][te++] ?? [0])[0],
+    triggers: technique[0][te++][0],
+    toughness_: technique[0][te++][0],
+    breakDmg: technique[0][te++][0],
+    maxBlessings: technique[0][te++][0],
+  },
+  b1: {
+    break_: bonusAbility1[0][0][0],
+  },
+  b2: {
+    energy: bonusAbility2[0][0][0],
+  },
+  b3: {
+    breakThreshold: bonusAbility3[0][0][0],
+    breakPer: bonusAbility3[0][0][1],
+    dmg_per: bonusAbility3[0][0][2],
+    max_dmg_: bonusAbility3[0][0][3],
   },
   e1: {
-    dmg: eidolon[1][0] ?? 0,
+    defIgn_: eidolon[1][0],
+  },
+  e2: {
+    atk_: eidolon[2][0],
+  },
+  e4: {
+    break_: eidolon[4][0],
+    duration: eidolon[4][1],
+  },
+  e6: {
+    ultDurationInc: eidolon[6][0],
+    breakDmgMult_inc: eidolon[6][1],
   },
 } as const
 
 const { char } = self
 
 // TODO: Add conditionals
-const { boolConditional } = allBoolConditionals(key)
-const { listConditional } = allListConditionals(key, ['val1', 'val2'])
-const { numConditional } = allNumConditionals(key, true, 0, 2)
+const { skillOvertone, ultZone, e4Broken } = allBoolConditionals(key)
+
+const e6TechniqueAddMult = cmpGE(char.eidolon, 6, dm.e6.breakDmgMult_inc)
 
 const sheet = register(
   key,
   // Handles base stats, StatBoosts and Eidolon 3 + 5
   entriesForChar(data_gen),
 
-  // TODO: Add formulas/buffs
   // Formulas
   ...dmg('basicDmg', baseTag, 'atk', dm.basic.dmg, 'basic'),
+  ...customBreakDmg(
+    'zoneBreakDmg',
+    { ...baseTag, damageType1: 'break' },
+    subscript(char.ult, dm.ult.breakDmg)
+  ),
+  ...customBreakDmg(
+    'talentBreakDmg',
+    { ...baseTag, damageType1: 'break' },
+    subscript(char.talent, dm.talent.breakDmg)
+  ),
+  ...customBreakDmg(
+    'techBreakDmg',
+    { ...baseTag, damageType1: 'break' },
+    sum(dm.technique.breakDmg, e6TechniqueAddMult)
+  ),
 
   // Buffs
-  selfBuff.premod.dmg_.add(cmpGE(char.eidolon, 6, boolConditional.ifOn(1))),
-  teamBuff.premod.dmg_.add(listConditional.map({ val1: 1, val2: 2 })),
-  enemyDebuff.common.defIgn_.add(numConditional)
+  teamBuff.premod.dmg_.add(
+    skillOvertone.ifOn(subscript(char.skill, dm.skill.dmg_))
+  ),
+  teamBuff.premod.weakness_.add(skillOvertone.ifOn(dm.skill.weakness_)),
+  teamBuff.premod.resPen_.add(
+    ultZone.ifOn(subscript(char.ult, dm.ult.resPen_))
+  ),
+  notSelfBuff.premod.spd_.add(subscript(char.talent, dm.talent.spd_)),
+  teamBuff.premod.brEff_.add(
+    cmpEq(char.bonusAbility1, 1, constant(dm.b1.break_))
+  ),
+  teamBuff.premod.dmg_.add(
+    cmpEq(
+      char.bonusAbility3,
+      1,
+      skillOvertone.ifOn(
+        min(
+          // (brEff_ - 1.2) / 0.1 * 0.06
+          prod(
+            sum(self.final.brEff_, -dm.b3.breakThreshold),
+            1 / dm.b3.breakPer,
+            dm.b3.breakPer
+          ),
+          constant(dm.b3.max_dmg_)
```

Comment:
Bare `dm.b3.max_dmg_` should work here (assuming it's has the correct type `number`)